template/src/api/routes/route.js

```diff
@@ -2,112 +2,100 @@
-  const subscribeMessage = channel.subscribe().message(0);
+  const operationId = operation.id();
+  const message = operation.messages().all()[0];
+  const messageValidationLogic = (operation.messages().length > 1) ? `
+  let nValidated = 0;
+  // For oneOf, only one message schema should match.
+  // Validate payload against each message and count those which validate
+
+  ${
+    operation.messages().all().map(message => `try {
+      nValidated = await validateMessage(message.payload,'${ channelAddress }','${ message.name() }','subscribe', nValidated);
+    } catch { };`).join('\n')
+  }
+
+  if (nValidated === 1) {
+    await ${camelCase(channelName)}Handler._${operationId}({message});
+    next()
+  } else {
+    throw new Error(\`\${nValidated} of ${ operation.messages().length } message schemas matched when exactly 1 should match\`);
+  }` : `await validateMessage(message.payload,'${ channelAddress }','${ message.name() }','subscribe');`;
   
   return `
-  ${channel.subscribe().summary()  ? `
+  ${operation.hasSummary()  ? `
   /**
-   * ${ channel.subscribe().summary() }
+   * ${ operation.summary() }
    */
   `: ''}
-  router.use('${toHermesTopic(channelName)}', async (message, next) => {
+  router.useOutbound('${toHermesTopic(channelAddress)}', async (message, next) => {
     try {
-      ${channel.subscribe().hasMultipleMessages() 
-        ? `
-      let nValidated = 0;
-      // For oneOf, only one message schema should match.
-      // Validate payload against each message and count those which validate
-
-      ${
-        Array.from(Array(channel.subscribe().messages().length).keys()).map(i => `try {
-          nValidated = await validateMessage(message.payload,'${ channelName }','${ channel.subscribe().message(i).name() }','subscribe', nValidated);
-        } catch { };`).join('\n')
-      }
-
-      if (nValidated === 1) {
-        await ${camelCase(channelName)}Handler._${subscribeOperationId}({message});
-        next()
-      } else {
-        throw new Error(\`\${nValidated} of ${ channel.subscribe().messages().length } message schemas matched when exactly 1 should match\`);
-      }
-        `
-        : `
-      await validateMessage(message.payload,'${ channelName }','${ subscribeMessage.name() }','subscribe');
-      await ${camelCase(channelName)}Handler._${ subscribeOperationId }({message});
+      ${isSpecV3 ? '' : messageValidationLogic}
+      await ${camelCase(channelName)}Handler._${ operationId }({message});
       next();
-        `
-      }
     } catch (e) {
       next(e);
     }
   });
   `;
 }
 
-function routeCode(channel, channelName) {
-  const hasPublish = channel.publish();
-  const hasSubscribe = channel.hasSubscribe();
-
+function routeCode(channel, isSpecV3) {
+  const channelName = channel.id();
   const generalImport = `
   const Router = require('hermesjs/lib/router');
   const { validateMessage } = require('../../lib/message-validator');
```

Comment:
this part also needs `isSpecV3` condition as we should not import not used library

also file `message-validator` should have a condition and not be rendered if v3 is used - but that can be left for later once we fix https://github.com/asyncapi/generator/issues/1206